src/common/plugin/Artifacts.js

```diff
@@ -0,0 +1,47 @@
+/*globals define*/
+define([
+], function(
+) {
+    const Artifacts = function () {};
+
+    // Supposed to be used as an extension object, do not instantiate
+    Artifacts.prototype.constructor = null;
+
+    Artifacts.prototype.getArtifactsDir = async function() {
+        // Find the artifacts dir
+        const children = await this.core.loadChildren(this.rootNode);
+        return children.find(child => this.core.getAttribute(child, 'name') === 'MyArtifacts') ||
+            this.activeNode;
+    };
+
+    Artifacts.prototype.getAssetNameFromHash = async function(hash){
```

Comment:
Same comment as below. This isn't actually shared between both plugins - why isn't it defined in the UploadArtifact plugin?

Reply:
Moved to child class.